### PR TITLE
sort variations options sku selector

### PR DESCRIPTION
**Pull request by rebecastilho**
#### What problem is this solving?

sku sorting selector that natively does not sort by size

#### How to test it?
https://www.jaked.com/ww/active-sport-jksc6kj01u/p?skuId=4112
http://rebeca--jaked.myvtex.com/

#### Screenshots or example usage:
Before 

image
After 

image

### CODE REVIEW

**Comment by rebecastilho** [comment]
> Hi! I'm VTEX IO CI/CD Bot and I'll be helping you to publish your app! 🤖
> 
> Please select which version do you want to release:
> 
> * [x ]  Patch (backwards-compatible bug fixes)
> * [ ]  Minor (backwards-compatible functionality)
> * [ ]  Major (incompatible API changes)
> 
> And then you just need to merge your PR when you are ready! There is **no need** to create a release commit/tag.
> 
> * [ ]  No thanks, I would rather do it manually 😞

**Comment by feraraujofilho** [comment]
@icazevedo any news here? We have multiple clients requesting it, Miriade, Yamamay, v73

File: react/components/SKUSelector/components/SKUSelector.tsx [hunk 1]
```diff
@@ -233,7 +233,10 @@ const variationNameToDisplayVariation = ({
       })
     )
     .filter(Boolean) as DisplayOption[]
-
+    
+    options.sort((a,b)=> {
+      return a.label < b.label ? -1 : a.label > b.label ? 1 : 0;
+    })
```
Comment (danzanzini) [inline react/components/SKUSelector/components/SKUSelector.tsx:239]: If you want to proceed with the merge, the sort must happen only when specified by a prop. This way, it will not break the behavior of other stores.

File: react/components/SKUSelector/components/SKUSelector.tsx [hunk 2]
```diff
@@ -248,8 +251,8 @@ const getAvailableVariations = ({
   disableUnavailableSelectOptions,
 }: AvailableVariationParams): DisplayVariation[] => {
   const variationCount = Object.keys(variations).length
-
-  return Object.keys(variations).map(
+ 
+  return Object.keys(variations).sort().map(
```
Comment (danzanzini) [inline react/components/SKUSelector/components/SKUSelector.tsx:255]: Can you clarify why you're sorting it twice?

**Review by danzanzini — COMMENTED** [review]
Hello, @rebecastilho. Sorry for taking this long to answer and thanks for contributing with us :)

I've checked the production link you sent and the sorting is good over there. I've checked other products and all of them seem to be working fine.

Is this still needed? If yes, I've suggested some improvements on code that needs to be tackled before proceeding with the merge.

Let me know if you need any assistance

**Comment by rebecastilho** [comment]
> Hello, @rebecastilho. Sorry for taking this long to answer and thanks for contributing with us :)
> 
> I've checked the production link you sent and the sorting is good over there. I've checked other products and all of them seem to be working fine.
> 
> Is this still needed? If yes, I've suggested some improvements on code that needs to be tackled before proceeding with the merge.
> 
> Let me know if you need any assistance

Hi [danzanzini](https://github.com/danzanzini), because of the delay I had to make a workaround in this store but we have more stores complaining about the same problem so I believe it is still necessary, thank you very much for the considerations, I made the changes.

File: react/components/SKUSelector/components/SKUSelector.tsx [hunk 3]
```diff
@@ -330,6 +343,7 @@ function SKUSelector({
       skuItems,
       hideImpossibleCombinations,
       disableUnavailableSelectOptions,
+      isSort
```
Comment (danzanzini) [inline react/components/SKUSelector/components/SKUSelector.tsx:346]:
```suggestion
      isSort,
```

File: react/components/SKUSelector/index.tsx [hunk 4]
```diff
@@ -186,6 +186,7 @@ interface Props {
   sliderDisplayThreshold?: number
   sliderArrowSize?: number
   sliderItemsPerPage?: ResponsiveValuesTypes.ResponsiveValue<number>
+  isSort:boolean
```
Comment (danzanzini) [inline react/components/SKUSelector/index.tsx:189]:
```suggestion
  isSort?: boolean
```

File: react/components/SKUSelector/components/SKUSelector.tsx [hunk 5]
```diff
@@ -234,6 +238,11 @@ const variationNameToDisplayVariation = ({
     )
     .filter(Boolean) as DisplayOption[]
 
+    if(isSort) {
+    options.sort((a,b)=> {
+      return a.label < b.label ? -1 : a.label > b.label ? 1 : 0;
+    })
+  }
```
Comment (danzanzini) [inline react/components/SKUSelector/components/SKUSelector.tsx:245]:
```suggestion
    if(isSort) {
      options.sort((a,b)=> {
        return a.label < b.label ? -1 : a.label > b.label ? 1 : 0;
      })
  }
```

File: docs/SKUSelector.md [hunk 6]
```diff
@@ -49,6 +49,7 @@ The `sku-selector` block is mainly used in Product Details Pages (PDPs) to displ
...
 | `displayMode` | `enum` | Defines how the product variation names will be displayed (it doesn't apply to product variation images). Possible values are: `default` (displays all variation names), `select` (only displays the selected variation name) or `slider` (displays all variation names in a slider when the number of available options in greater than the value defined in the `sliderDisplayThreshold` prop). Notice that this prop is _responsive_, so you can declare an object as its value specifying a value for each breakpoint (`desktop` and `mobile`). | `default`  |
 | `hideImpossibleCombinations`     | `boolean`   | Defines if a product variation should be clickable (`true`) or not (`false`). If `true`, the variation won't be displayed. If `false`, the variation is displayed with less opacity. For example, there are two colors for a pair of sneakers: pink and white, and if the pink sneaker is available for only one size, the other sizes won't be displayed (`true`).| `true`  |
+| `isSort`     | `boolean`   | Defines if the variations are to be sorted in alphabetical order.| `false`  |
```
Comment (danzanzini) [inline docs/SKUSelector.md:52]: What do you think of a more explicitly name like `sortVariationsByLabel` or `sortByLabel`? I think it will make the code clearer

**Review by danzanzini — CHANGES_REQUESTED** [review]
Hi @rebecastilho. Thanks for the changes. 

I've made some comments on code, suggesting changes following our code pattern. Do you use any lint tools? If not, I recommend looking at tools like [ESLint](https://eslint.org/). They make our life a lot easier 😄 

The link you provided when opening the PR has these new changes to test?

**Comment by rebecastilho** [comment]
> Hi @rebecastilho, thanks for the changes. Now it seems excellent, but I will have to ask you one more thing. 😬
> 
> Can you write some specs for it? We already have tests for this component at https://github.com/vtex-apps/store-components/blob/master/react/__tests__/components/SKUSelector.test.tsx. You just need to add one more test to ensure the order when this prop is set to true.
> 
> Thanks again for the contribution. Please, ask for my review again when it's done so I will not take to long to review after the changes.

Sorry for the delay, I didn't see your point, I added the test, thanks for the reviews :)

File: react/__tests__/components/SKUSelector.test.tsx [hunk 7]
```diff
@@ -1408,4 +1408,55 @@ describe('<SKUSelector />', () => {
 
     expect(queryByText('Yellow')).toBeFalsy()
   })
+
```
Comment (danzanzini) [inline react/__tests__/components/SKUSelector.test.tsx:1411]:
```suggestion

```

File: react/__tests__/components/SKUSelector.test.tsx [hunk 8]
```diff
@@ -1408,4 +1408,55 @@ describe('<SKUSelector />', () => {
 
     expect(queryByText('Yellow')).toBeFalsy()
   })
+  
+  it('must order sku specification to be sorted in alphabetical order', async () => {
+    const defaultSeller =  {sellerDefault: true, commertialOffer: { Price: 1, ListPrice: 1, AvailableQuantity: 2 }}
```
Comment (danzanzini) [inline react/__tests__/components/SKUSelector.test.tsx:1413]:
```suggestion
    const defaultSeller =  {
      sellerDefault: true, 
      commertialOffer: { Price: 1, ListPrice: 1, AvailableQuantity: 2 },
    }
```

File: react/__tests__/components/SKUSelector.test.tsx [hunk 9]
```diff
@@ -1408,4 +1408,55 @@ describe('<SKUSelector />', () => {
...
+            values: ['41'],
+          },
+          { name: 'Color', values: ['Gray'] },
+        ],
+        sellers: [defaultSeller],
+        images: [],
+      },
+    ]
+
+    const {container} =  renderComponent({ sortVariationsByLabel:true,skuSelected:skuItems[0], skuItems:skuItems})
```
Comment (danzanzini) [inline react/__tests__/components/SKUSelector.test.tsx:1454]:
```suggestion
    const { container } = renderComponent({
      sortVariationsByLabel: true,
      skuSelected: skuItems[0], 
      skuItems:skuItems,
    })
```

File: react/__tests__/components/SKUSelector.test.tsx [hunk 10]
```diff
@@ -1408,4 +1408,55 @@ describe('<SKUSelector />', () => {
...
+    
+    const options = container.querySelectorAll('.skuSelectorItemTextValue')
+    expect(options[0]).toHaveTextContent('41')
+    expect(options[1]).toHaveTextContent('42')
+    expect(options[2]).toHaveTextContent('43')
+    })
```
Comment (danzanzini) [inline react/__tests__/components/SKUSelector.test.tsx:1460]:
```suggestion
  })
```

File: react/__tests__/components/SKUSelector.test.tsx [hunk 11]
```diff
@@ -1408,4 +1408,55 @@ describe('<SKUSelector />', () => {
...
 })
+
```
Comment (danzanzini) [inline react/__tests__/components/SKUSelector.test.tsx:1462]:
```suggestion
```

**Review by danzanzini — CHANGES_REQUESTED** [review]
Hi @rebecastilho. Thanks for doing the tests. It's very important to us :)

We had some problems with the Lint test. They're all detailed in the `Files Changed` of the Pull Request. 

I've made some suggestions to fix some of them. Can you apply them and fix any remaining problems, please?

**Review by danzanzini — APPROVED** [review]
Hi @rebecastilho. I think now it's done. 👏 🎉

I'll just check why the Danger CI is failing so I can merge and deploy the PR. Thanks for the contribution ⭐

File: package.json [hunk 12]
```diff
@@ -35,9 +31,12 @@
...
     "eslint-config-vtex": "^12.8.1",
     "eslint-config-vtex-react": "^6.7.3",
-    "husky": "^4.2.0",
+    "husky": "^7.0.1",
     "lint-staged": "^10.0.2",
     "prettier": "2.3.2",
     "typescript": "^3.7.5"
+  },
+  "resolutions": {
+    "glob-parent": "^6.0.1"
```
Comment (danzanzini) [inline package.json:40]: Why are these changes needed?

File: package.json [hunk 13]
```diff
@@ -8,12 +8,8 @@
     "format": "prettier --write \"**/*.{ts,tsx,js,jsx,json}\"",
     "test": "cd ./react && yarn test",
     "verify": "yarn lint && yarn test",
-    "lint:locales": "intl-equalizer"
-  },
-  "husky": {
-    "hooks": {
-      "pre-commit": "lint-staged"
-    }
```
Comment (emersonlaurentino) [inline package.json:16]: you need add the hooks 
https://github.com/typicode/husky#usage

**Review by emersonlaurentino — CHANGES_REQUESTED** [review]

File: package.json (hunk 12 above)
Comment (lariciamota) [reply package.json:40]: @rebecastilho I think it needs a rebase, then those unrelated changes won't appear in this PR

File: package.json (hunk 13 above)
Comment (danzanzini) [reply package.json:16]: No, it just needs a rebase as @lariciamota said above. The changes on husky were done at other PR

Comment (emersonlaurentino) [reply package.json:16]: Hm, okay. So, someone need to make the rebase

**Review by emersonlaurentino — APPROVED** [review]

**Comment by danzanzini** [comment]
Merged, tested [here](https://danzan--storecomponents.myvtex.com/working-shirt/p?skuId=2000548), and deployed. 🚀 

@rebecastilho Thanks for your contribution. Since this is not a breaking change, this update will be automatically installed at your store.
